ember-headless-form/src/components/-private/control/input.ts

```diff
@@ -28,35 +28,37 @@ export type InputType =
   | 'url'
   | 'week';
 
-export interface HeadlessFormControlInputComponentSignature<VALUE> {
+export interface HeadlessFormControlInputComponentSignature {
   Element: HTMLInputElement;
   Args: {
-    value: VALUE;
+    value: string;
     type?: InputType;
     fieldId: string;
-    setValue: (value: VALUE) => void;
+    setValue: (value: string) => void;
   };
 }
 
-export default class HeadlessFormControlInputComponent<VALUE> extends Component<
-  HeadlessFormControlInputComponentSignature<VALUE>
-> {
-  get type(): InputType {
-    return this.args.type ?? 'text';
-  }
-
-  get valueAsString(): string {
+export default class HeadlessFormControlInputComponent extends Component<HeadlessFormControlInputComponentSignature> {
+  constructor(
+    owner: unknown,
+    args: HeadlessFormControlInputComponentSignature['Args']
+  ) {
     assert(
-      `input can only handle string values, but you passed ${typeof this.args
-        .value}`,
-      typeof this.args.value === 'string'
+      `input component does not support @type="${args.type}" as there is a dedicated component for this. Please use the \`field.${args.type}\` instead!`,
+      args.type === undefined ||
+        // TS would guard us against using an unsupported `InputType`, but for JS consumers we add a dev-only runtime check here
+        (args.type as string) !== 'checkbox'
     );
 
-    return this.args.value;
+    super(owner, args);
+  }
+
+  get type(): InputType {
+    return this.args.type ?? 'text';
```

Comment:
Question, nit: is there a way to set the `args.type` default as `text`? Just wondering if there is a way to hint to the user that `text` is the default type without them having to scroll down to this getter.

Reply:
not yet, but there is a followup proposal to `<template>` that adds exactly this

Reply:
> but there is a followup proposal to <template> that adds exactly this

Uh, which is this? Have a link at hand?

Reply:
default args to `<template>`, I think the ideal is something like
```js
<template signature={TheSignature} defaultArgs={defaults}>

</template>
```
this _greatly_ reduces boilerplate for template-only components, too.
No link yet, as it's not written. Not sure of @chriskrycho or @dfreeman are going to write it, or maybe it's too early for that kind of planning atm, as they're focused on v1 of Glint atm.